object_store/src/lib.rs

```diff
@@ -1685,8 +1702,86 @@ mod tests {
         assert!(stream.next().await.is_none());
     }
 
-    // Tests TODO:
-    // GET nonexisting location (in_memory/file)
-    // DELETE nonexisting location
-    // PUT overwriting
+    #[test]
+    fn test_get_options() {
```

Comment:
This is really a test of `check_preconditions` isn't it?

```suggestion
    fn test_pre_conditions() {
```